### PR TITLE
Add configuration option to ignore unknown arguments

### PR DESCRIPTION
**Pull request by jamesnetherton**
I set the default behaviour to throw an exception if unknown args are found.

Maybe the default should be to log a warning instead?

### CODE REVIEW

File: extensions-core/main/runtime/src/main/java/org/apache/camel/quarkus/main/CamelMainConfig.java [hunk 1]
```diff
@@ -40,4 +46,27 @@
...
+     *
+     * FAIL - Prints the {@link CamelMain} usage statement and throws a {@link RuntimeException}
+     * IGNORE - Suppresses any warnings and the application startup proceeds as normal
+     * WARN - Prints the {@link CamelMain} usage statement but allows the application startup to proceed as normal
+     */
+    public enum UnknownArgumentFailMode {
+        FAIL,
+        IGNORE,
+        WARN
+    }
```
Comment (ppalaga) [inline extensions-core/main/runtime/src/main/java/org/apache/camel/quarkus/main/CamelMainConfig.java:71]: There is `org.apache.camel.quarkus.core.CamelConfig.FailureRemedy` which I assume is visible here. Shouldn't we re-use it?  
Wouldn't the `ArgumentConfig.onUnknown` field be a better place for the FAIL/IGNORE/WARN JavaDoc? - in that way user could see it on the extension page.

**Review by ppalaga — COMMENTED** [review]
Nice test coverage, thanks! One suggestion inline.

File: extensions-core/main/runtime/src/main/java/org/apache/camel/quarkus/main/CamelMainConfig.java (hunk 1 above)
Comment (jamesnetherton) [reply extensions-core/main/runtime/src/main/java/org/apache/camel/quarkus/main/CamelMainConfig.java]: Latest commit applies those suggestions.

**Review by oscerd — APPROVED** [review]

**Comment by ppalaga** [comment]
> should be to log a warning instead?

I am not going to veto, if Luca favors that :)

**Review by ppalaga — APPROVED** [review]

**Comment by jamesnetherton** [comment]
> I am not going to veto, if Luca favors that :)

I think it's actually my preference too. The warning is big and ugly enough that folks should hopefully notice something is wrong.

**Comment by lburgazzoli** [comment]
Ah, I did it from the mobile app and it was mainly for the PR per se not specifically on the warn vs exception so @ppalaga you can put your veto :)

**Comment by jamesnetherton** [comment]
Latest commit changes the default to `warn`.

**Pull request merged**